prime-dotnet-webapi/Services/EmailService.cs

```diff
@@ -342,6 +343,23 @@ public async Task SendPaperEnrolmentSubmissionEmailAsync(int enrolleeId)
             await Send(email);
         }
 
+        public async Task SendEnrolleeAbsenceNotificationEmailAsync(int enrolleeId, EnrolleeAbsenceViewModel absence, string email)
+        {
+            var enrolleeDto = await _context.Enrollees
+                .Where(e => e.Id == enrolleeId)
+                .Select(e => new
+                {
+                    e.FirstName,
+                    e.LastName
+                })
+                .SingleOrDefaultAsync();
+
+            var viewModel = new EnrolleeAbsenceNotificationEmailViewModel(enrolleeDto.FirstName, enrolleeDto.LastName, absence.StartTimestamp, absence.EndTimestamp);
```

Comment:
It seems the intermediate `enrolleeDto` is not necessary as the `viewModel` could be constructed directly in the LINQ query. 

Further, in theory `enrolleeDto` could be null by calling `SingleOrDefaultAsync` hence causing null exception in `enrolleeDto.FirstName` and `enrolleeDto.LastName` later.
```C#
var viewModel = await _context.Enrollees
    .Where(e => e.Id == enrolleeId)
    .Select(e => new EnrolleeAbsenceNotificationEmailViewModel(
        e.FirstName, e.LastName, absence.StartTimestamp, absence.EndTimestamp)
    )
    .SingleOrDefaultAsync();
```